### PR TITLE
E2E Tests: Fix withExperimentalFeatures util

### LINKED ISSUE #10916: E2E Tests: Fix `withExperimentalFeatures` util
## Task Description

As noted in #10808 and also before that, `withExperimentalFeatures` doesn't currently work when actually passing more than 1 feature flag. It appears that the second checkbox doesn't get properly checked.

See also #7748 where `withExperimentalFeatures` has been changed for similar reasons. Maybe that change there fixes this already.

### PR DESCRIPTION
**Pull request by timarney**
## Context

`withExperimentalFeatures` currently accepts an array of "features". Looping over those feature to enable/disable  `experiments` isn't working as expected.

PR #10808 works around this by calling `withExperimentalFeatures` multiples times vs using the array format.

## Summary

- This PR updates how we loop over items in `withExperimentalFeatures` 
- It updates the **Font Check test** to use the **array** feature format

## Relevant Technical Choices

Noting: 
- This PR includes some updates from #7748

https://github.com/GoogleForCreators/web-stories-wp/pull/7748/files#diff-043d824399de44fd91ca884ab3d8564fbde22f0bc88138c0b9282c1c897d6641L43

- I have moved `removeAllFonts` to the custom fonts utils so it can be re-used between the custom fonts tests and the font check tests 

**Current:**
```javascript
withExperimentalFeatures(['customFonts']);
withExperimentalFeatures(['notifyDeletedFonts']); 
```

**Updated**

```javascript
withExperimentalFeatures(['customFonts', 'notifyDeletedFonts']);
```

## User-facing changes
N/A

## Testing Instructions

- Tested using e2e test suite

### Does this PR have a security-related impact?

No

### Does this PR change what data or activity we track or use?

No

### Does this PR have a legal-related impact?

No

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [ ] I have verified accessibility to the best of my abilities ([docs](https://github.com/googleforcreators/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [ ] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/googleforcreators/web-stories-wp/tree/main/docs#testing))
- [ ] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

Fixes #10916

<hr>
<hr>

**Testing pre updates**

👉 👉  👉  **Grabbing some snapshots for tracking purposes** 

https://github.com/GoogleForCreators/web-stories-wp/pull/7748/files#diff-043d824399de44fd91ca884ab3d8564fbde22f0bc88138c0b9282c1c897d6641L43

For this test:

**Locally --- enabling --- looks fine:**

**Percy**

https://percy.io/google/web-stories-wp-e2e/builds/16681165/changed/937869660?browser=chrome&browser_ids=23&subcategories=unreviewed%2Cchanges_requested&viewLayout=side-by-side&viewMode=new&width=1600&widths=1600

**Locally --- disabling--- ❌  :**

Looks like the custom fonts checkbox doesn't get unchecked here:

**Percy**
https://percy.io/google/web-stories-wp-e2e/builds/16681165/changed/937869363?browser=chrome&browser_ids=23&subcategories=unreviewed%2Cchanges_requested&viewLayout=side-by-side&viewMode=new&width=1600&widths=1600

### CODE REVIEW

**Comment by spacedmonkey** [comment]
Does this fix this - https://github.com/googleforcreators/web-stories-wp/issues/10916

**Comment by timarney** [comment]
@spacedmonkey 

> Does this fix this - https://github.com/GoogleForCreators/web-stories-wp/issues/10916b

Tested locally --- and seems to.

Pushed a commit.

**Comment by timarney** [comment]
Grabbing some more screens

File: packages/e2e-test-utils/src/customFonts.js [hunk 1]
```diff
@@ -13,13 +13,30 @@
...
  */
+
 function parseText(optionsText) {
   return optionsText.map((option) => {
     const [name, url] = option.split('\n');
     return { name, url };
   });
 }
 
+export const removeAllFonts = async () => {
```
Comment (timarney) [inline packages/e2e-test-utils/src/customFonts.js:24]: Code from #10918 just moving to `utils` for re-use

File: packages/e2e-test-utils/src/experimentalFeatures.js [hunk 2]
```diff
@@ -36,23 +36,26 @@ export async function toggleExperiments(features, enable) {
...
-      if ((!isChecked && enable) || (isChecked && !enable)) {
-        await page.click(selector);
-      }
-    })
-  );
+  /* eslint-disable no-await-in-loop */
+  for (let i = 0; i < features.length; i++) {
+    const selector = `#${features[i]}`;
+    await page.waitForSelector(selector);
+    const checkedSelector = `${selector}:checked`;
```
Comment (timarney) [inline packages/e2e-test-utils/src/experimentalFeatures.js:43]: Via #7748

File: packages/e2e-test-utils/src/experimentalFeatures.js [hunk 3]
```diff
@@ -36,23 +36,26 @@ export async function toggleExperiments(features, enable) {
...
+      await page.click(selector);
+    }
+  }
+  /* eslint-enable no-await-in-loop */
 
-  await Promise.all([
-    page.waitForNavigation({ waitUntil: 'networkidle0' }),
-    page.click('#submit'),
-  ]);
+  // Ensures the button is visible and can be clicked on in Firefox.
```
Comment (timarney) [inline packages/e2e-test-utils/src/experimentalFeatures.js:52]: Via #7748

File: packages/e2e-tests/src/specs/dashboard/settings/adminUser/customFonts.js [hunk 4]
```diff
@@ -36,23 +36,6 @@ const OPEN_SANS_CONDENSED_LIGHT_ITALIC_URL = `${FONT_BASE_URL}/OpenSansCondensed
 
 const findByUrl = (arr, val) => arr.find((o) => o.url === val);
 
-const removeAllFonts = async () => {
```
Comment (timarney) [inline packages/e2e-tests/src/specs/dashboard/settings/adminUser/customFonts.js:44]: Moved to util above

File: packages/e2e-tests/src/specs/editor/fontCheck/fontCheck.js [hunk 5]
```diff
@@ -87,9 +88,12 @@ async function storyWithFontCheckDialog(title) {
 }
 
 describe('Font Check', () => {
-  // TODO(#10916): Combine these calls.
-  withExperimentalFeatures(['customFonts']);
-  withExperimentalFeatures(['notifyDeletedFonts']);
+  withExperimentalFeatures(['customFonts', 'notifyDeletedFonts']);
```
Comment (timarney) [inline packages/e2e-tests/src/specs/editor/fontCheck/fontCheck.js:91]: Fixes @todo

File: packages/e2e-tests/src/specs/editor/fontCheck/fontCheck.js [hunk 6]
```diff
@@ -87,9 +88,12 @@ async function storyWithFontCheckDialog(title) {
...
+
+  beforeAll(async () => {
+    await visitSettings();
```
Comment (timarney) [inline packages/e2e-tests/src/specs/editor/fontCheck/fontCheck.js:94]: This test was missing was missing cleanup

**Review by swissspidy — APPROVED** [review]
Looks good 👍

**Pull request merged**